src/App.tsx

```diff
@@ -201,7 +262,25 @@ class Numpad extends React.Component<NumpadProps, {}> {
     }
 
     public renderNumButton(i: SquareValue) {
-        return <Button onClick={() => {this.props.onClickNum(i); }} className="button-num" variant="contained">{i as number}</Button>;
+        if (this.props.numpadMode == "normal" as NumberMode) {
+            return <Button
+                onClick={() => {
+                    this.props.onClickNorm(i);
+                }}
+                className="button-num"
+                variant="contained">
+                    {i as number}
+            </Button>;
+        } else {
+            return <Button
+                onClick={() => {
+                    this.props.onClickCorner(i);
```

Comment:
The only difference between these two if statements is which function you call. Replace this with:

`onClick={() => this.props.numpadMode == "normal" ? this.props.onClickNormal(i) : this.props.onClickCorner(i)}`